controllers/networking-calico/pkg/controller/actuator_reconcile.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	resourcemanagerv1alpha1 "github.com/gardener/gardener-resource-manager/pkg/apis/resources/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	resourcemanager "github.com/gardener/gardener-resource-manager/pkg/manager"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/gardener/gardener-extensions/controllers/networking-calico/pkg/charts"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+)
+
+const (
+	calicoSecretName       = "calico"
+	calicoConfigSecretName = "calico-config"
+	calicoRBACSecretName   = "calico-rbac"
+	calicoPSPsSecretName   = "calico-psps"
+
+	noCleanUp = "shoot.gardener.cloud/no-cleanup"
+)
+
+func withYamlSeparator(str string) string {
+	return str + "\n" + "---" + "\n"
+}
+
+func withLocalObjectRefs(refs ...string) []corev1.LocalObjectReference {
+	var localObjectRefs []corev1.LocalObjectReference
+	for _, ref := range refs {
+		localObjectRefs = append(localObjectRefs, corev1.LocalObjectReference{Name: ref})
+	}
+	return localObjectRefs
+}
+
+func Secrets(cl client.Client, calicoFiles *charts.CalicoFiles, namespace string) (*resourcemanager.Secrets, []corev1.LocalObjectReference) {
```

Comment:
TBH, I really don't like this in particular. As discussed yesterday we will provide better mechanisms to check the health/status of the managed resources. Having this makes it very hard to update/change anything, it's a lot of complexity, and code that will be deleted for sure anyways very soon. I suggest to remove all of this already now such that we don't need to maintain it.

Reply:
I am not getting it. The Secrets functions just creates the secrets required for holding the content of the Managed resources, or do you mean the file created methods?

Reply:
Oops, the comment was meant for another code part (little further down). I just added it there.
What I mean here is: What's the benefit is to categorize these resources into different files? Why not go the easiest way, create a `ManagedResource` with exactly one `secretRef` that contains all calico related files? What's the benefit of splitting this up into multiple secrets?

Reply:
This way it maps to the helm chart structure, so looking at the managed resource spec, we can directly map and locate files in the helm chart. I agree that it is not needed, and throwing in a blob would be easier, but it makes clearer to see also easier for debugging when something is wrong, we just look at the related secret.

Reply:
Well, do you have better/stronger arguments than 

>  so looking at the managed resource spec, we can directly map and locate files in the helm chart.

? Maybe an example situation in which this makes things easier?
I don't think it makes anything easier. If it cannot apply a certain resource (maybe because of a syntax error, whatever) you first have to find the correct secret.
Considering the amount of code that also needs to be adapted if we update Calico and a new resource joins the game I would weight the "less work maintaining it and less effort debugging it" argument higher than the "directly maps to the helm chart" argument.